src/parser/type_checking.c

```diff
@@ -19964,7 +20006,6 @@ pt_fold_const_function (PARSER_CONTEXT * parser, PT_NODE * func)
 	      func->info.function.arg_list = NULL;
 	    }
 	}
-      func->type_enum = PT_TYPE_INTEGER;
```

Comment:
Is it not necessary to replace PT_TYPE_INTEGER with PT_TYPE_BIGINT?

Reply:
There are two routines to evaluate the (arg, return) type for the function node. I migrated _PT_COUNT_ and _PT_COUNT_STAR_ into the new routine. The new routine (_pt_eval_function_type_new ()_) gets type of arguments and return type information from **func_all_signatures** struct defined at func_type.cpp file. From now, the count and count star function will not reach to here. That's why I put the _assert(false)_ here.

Reply:
And when doing a constant folding for spacial cases (COUNT(1), COUNT(:x)... described at the comment above), there is no reason to set a return type here If there is nothing changed. Even this line is out of if-statement (it always set a return type).
So It seems a bug, I removed it.

Reply:
Thank you for your kind.